### ISSUE TITLE
Query on validation result

### ISSUE THREAD

**Issue by RokiaAbdeen** [Sayan98/pytorch-segnet#4]
when I have used the validation dataset to test the code I got a very bad and random result but I got an excellent result when I tested the model using the same training data
what would be the mistake and how to improve the segmentation performance for new unseen dataset?

**Comment by say4n** [comment]
You seem to be over-fitting to your training dataset, this should be
helpful:
https://machinelearningmastery.com/introduction-to-regularization-to-reduce-overfitting-and-improve-generalization-error/

Kudos! :)

**Comment by RokiaAbdeen** [comment]
I know every thing in this link about overfitting
I just need a quick solution , this is your code and you of course know how to solve overfitting in your code
can you tell me what would be the best solution for this overfitting?

**Comment by say4n** [comment]
Over-fitting is based on the cardinality of your dataset and the hyper parameters you use for training, so you'll have to figure them out on your own.
Also mate, please watch your tone. :)

**Comment by RokiaAbdeen** [comment]
I have used pascal voc 2007 for training and testing

**Comment by RokiaAbdeen** [comment]
I am sorry if any thing in my words is not ok.. really sorry
thanks alot this is your effort
thanks to help us with your code
I am student and I just need your help that's all :(

**Comment by RokiaAbdeen** [comment]
I just used the code as it is and used pascal voc 2007 for training and testing
and I found that all the tested images are segmented randomly, but it is doing very well with training data, I know that it is an overfitting problem but how to solve this problem in this code based on your experience, please..

**Comment by say4n** [comment]
Uhm, you could try reducing the learning rate, perhaps use early stopping or the likes. 

You can alter the [epochs, learning rate, momentum, batch size](https://github.com/say4n/pytorch-segnet/blob/be538aecd677b703e0cd9d803b51e33c06fa8a27/src/train.py#L29) etc. in [`src/train.py`](https://github.com/say4n/pytorch-segnet/blob/master/src/train.py).  6000 epochs seem kinda excessive now that I see it in the script! 

Also, you can implement early stopping by keeping a history of [`is_better`](https://github.com/say4n/pytorch-segnet/blob/be538aecd677b703e0cd9d803b51e33c06fa8a27/src/train.py#L84) and stop if say the last 10 epochs did not improve the performance. 

If you have any other doubts, do feel free to ask!

:)

**Comment by RokiaAbdeen** [comment]
Hello,
I have changed and tried different values of learning rate and momentum and I've saved many models after different number of epochs when the loss is decreased but it still giving random segmentation results which are not correct at all :(
this experiments took very long time can you help me to find solution, please..
